test/src/test/java/hudson/model/UpdateSiteTest.java

```diff
@@ -204,6 +206,25 @@ public void shutdownWebserver() throws Exception {
         assertEquals("TheHighTitle", available.get(1).getDisplayName());
     }
 
+    @Test public void deprecations() throws Exception {
+        UpdateSite site = getUpdateSite("/plugins/deprecations-update-center.json");
+
+        UpdateSite.Plugin credentials = site.getPlugin("credentials");
+        assertNotNull(credentials);
+        assertFalse(credentials.isDeprecated());
+        assertNull(credentials.getDeprecation());
+
+        UpdateSite.Plugin iconShim = site.getPlugin("icon-shim");
+        assertNotNull(iconShim);
+        assertTrue(iconShim.isDeprecated());
+        assertEquals("https://www.jenkins.io/deprecations/icon-shim/", iconShim.getDeprecation().url);
+
+        UpdateSite.Plugin tokenMacro = site.getPlugin("token-macro");
+        assertNotNull(tokenMacro);
+        assertTrue(tokenMacro.isDeprecated());
+        assertEquals("https://wiki.jenkins-ci.org/display/JENKINS/Token+Macro+Plugin", tokenMacro.getDeprecation().url);
```

Comment:
This loses test coverage for plugins not currently being published, but marked as deprecated via top-level metadata.

While we can assume that's kinda taken care of by the `icon-shim` test having the top-level deprecation URL, an explicit assertion would be nicer. Given `mandatory-depender`, `dependee`, etc. plugins are around, could they stand in for real plugins for this purpose?

FWIW while it's mostly true that

> the call to `getInfo()` was incidental to the primary purpose of the test and I did not deem it worth keeping.

it confirmed absence of the plugin from update site metadata.

Reply:
I added an assertion for a plugin not currently being published, but marked as deprecated via top-level metadata.